src/Chapter09.Tests/Listing09.15.RecordDeconstructorsAndPatternMatching.Tests.cs

```diff
@@ -0,0 +1,26 @@
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter09.Listing09_15.Tests;
+
+using AddisonWesley.Michaelis.EssentialCSharp.Chapter09.Listing09_04;
+
+
+[TestClass]
+public class PatternMatchingTest
+{
+    [TestMethod]
+    public void Deconstructor_CorrespondingParameters_EnablePatternMatching()
+    {
+        (int degrees, int minutes, int seconds, string name) = (
+            90, 0, 0, null);
+
+        // The constructor is generated using positional parameters
+        Angle angle = new(degrees, minutes, seconds, name);
```

Comment:
Though a clever way of doing multiple variable assignment. I think it might be clearer to simply pass these directly to the constructor.
```suggestion
        // The constructor is generated using positional parameters
        Angle angle = new(90, 0, 0, null);
```